lms/djangoapps/discussion/rest_api/permissions.py

```diff
@@ -63,11 +63,15 @@ def get_editable_fields(cc_content, context):
     # no edits, except 'abuse_flagged' and 'read' are allowed for thread
     # no edits, except 'abuse_flagged' is allowed for comment
     ret = {"abuse_flagged"}
-    if cc_content["type"] == "thread" and cc_content["closed"]:
-        ret |= {"read"}
-        return ret
-    if cc_content["type"] == "comment" and context["thread"]["closed"]:
-        return ret
+    if cc_content["type"] == "thread":
```

Comment:
Do you think it's worth refactoring `get_editable_fields` function so it looks something like this:
```python
if cc_content["type"] == "comment":
    closed, ret = get_editable_fields_comment(cc_content, context)
elif cc_content["type"] == "thread":
    closed, ret = get_editable_fields_thread(cc_content, context)

ret |= {"abuse_flagged"}

if not closed:
    # Shared fields

return ret
```
I think it can greatly reduce complexity and improve readability. I'm not entirely sure what are the downsides.